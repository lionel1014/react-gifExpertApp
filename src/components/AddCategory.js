import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');  // => ('') > () => ('') != () => ( ) da una error, envia una variale undefined

    const handleInputChange = (e) =>{//los valores que recibimos del input
        const {value} = e.target;
        setInputValue ( value );

        // console.log('handleInputChange fue llamado con exito')

    };

    const handleSubmit = (e) =>{//para enviar el input en un form
        e.preventDefault();
        
        if ( inputValue.trim().length >= 3 ){
            setCategories(cate => [inputValue,...cate]); // guardamos el nuevo valor del `inputValue`
            setInputValue (''); // el estado del inputValue ahora es vacío
        }

    }

    return (
        <form onSubmit = {handleSubmit} > 
            <p> {inputValue} </p>
            <input type="text" value={inputValue} placeholder='Ingresa una categoría'
            onChange = { handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {

    setCategories : PropTypes.func.isRequired

};
