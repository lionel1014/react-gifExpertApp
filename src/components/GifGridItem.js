import React from 'react';
import PropTypes from 'prop-types'; 

export const GifGridItem = ({title,url}) => {

    // console.log({title,url})

    return (
        <div className="imagenes animate__animated animate__flipInX" >
            <p> {title} </p>
            <img src={ url } alt= {title} />
        </div>
    )
}

GifGridItem.propTypes ={

    title : PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};
