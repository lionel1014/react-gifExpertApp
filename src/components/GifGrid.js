import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => { // no te olvides de desestructurar la propiedad

    const {data:images,loading} = useFetchGifs(category);  //data:images, en la desestructuracion cambiamos el nombre de la variable asi
    
    // useEffect(() =>{
    //     //importamos la promesa getGifs de "helpers"
    //     getGifs(category)
    //         .then(setImages)
    //         // .then( imgs => setImages(imgs)) // una funcion que manda el primer argumento al primer argumento de setImages
    //         // se lo puede resumir como el de arriba
    // },[category]); // que haga lo que hay dentro si cambia category

    return (
        <>
        <h3 className="animate__animated animate__bounceInLeft "> {category} </h3>
        {/* {loading ? 'Cargando...' : 'Fin de la carga del objeto.'} */}
         {loading && <p className="animate__animated animate__bounceIn " >Cargando...</p>}  {/*el && es una condición lógica AND */}
        
        <div className="imagenesGrid" >
            
            {
                images.map( ( img ) =>{
                    return <GifGridItem 
                            key = {img.id}
                            {...img} 
                            />
                }) 
            }
            
        </div>
        </>
    )
}
