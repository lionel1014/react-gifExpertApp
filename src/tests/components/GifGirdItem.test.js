import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
import React from 'react'


describe('Pruebas en GifGridItem => componente', () => {

    const title = 'Un título';
    const probando ='https://localhots/jajaja.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={probando} /> )
    
    test('Debe de mostrar el componente correctamente ', () => {
    
        expect(wrapper).toMatchSnapshot();

    });

    test('El title debe de tener un párrafo con un títilo', () => {
        
        const parrafo = wrapper.find('p');
        expect(parrafo.text().trim()).toBe(title);

    });
    
    test('Debe devolver una img con la "url" y "alt" de los props', () => {
        
        const img = wrapper.find('img');
        // console.log(img.prop('src'));
        expect(img.prop('src')).toBe(probando);
        expect(img.prop('alt')).toBe(title);

    });
    
    test('Debe de tener animate__flipInX en su "className" del <div>', () => {
        
        const div = wrapper.find( 'div' );

        // console.log(div.prop('className')); // me muestra el contenido de className

        const clase = div.prop('className')
        // console.log(clase.indexOf('animate__flipInX')); // esto extrae la ubicacion de donde se encuentra lo que buscamos
        // console.log(clase.trim().slice(27,)); // y esto reconstrulle a partir de (desde donde empieza,hasta donde termina) de extraer

        expect(clase.trim().slice(27,)).toBe('animate__flipInX'); //la que yo hice
        expect(clase.includes('animate__flipInX')).toBe(true); // preguntamos si clase incluye algo de esto

    })
    

})

