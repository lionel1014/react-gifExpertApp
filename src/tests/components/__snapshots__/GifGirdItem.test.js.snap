// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem => componente Debe de mostrar el componente correctamente  1`] = `
<div
  className="imagenes animate__animated animate__flipInX"
>
  <p>
     
    Un título
     
  </p>
  <img
    alt="Un título"
    src="https://localhots/jajaja.jpg"
  />
</div>
`;
